src/libstd/thread/local.rs

```diff
@@ -172,7 +172,8 @@ macro_rules! __thread_local_inner {
                 static __KEY: $crate::thread::__OsLocalKeyInner<$t> =
                     $crate::thread::__OsLocalKeyInner::new();
 
-                __KEY.get(__init)
+                #[allow(unused_unsafe)]
```

Comment:
I don't understand why the `allow` is needed.

Reply:
The macro is expanded inside the scope of the ˋ#![deny(...)]ˋ, causing an error when the allow is omitted. There are some messages about that in the original issue

Reply:
What will remind us to remove this `allow` when the call sites are fixed?

Reply:
Nothing I know of. As long as the unsafe-op-in-unsafe-fn feature is not both stable and a default, we will need it to allow for both use cases of activating/not activating the feature

Reply:
Hm, I would expect the lint to not propagate into macros, can you file an issue for that? It should probably allow itself inside macros (unless the macro itself denies it).

Reply:
See [this playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=8f01524ce1e2668692bc4a7eb4af473c). When activating `#![deny(unsafe_op_in_unsafe_fn)]`, their will be conflicts between places that uses it and places that don't, meaning there is either a missing `unsafe {}` block or an extraneous one. For now, using `#[allow(unused_unsafe)]` is the only solution I know of.

Reply:
Yeah this allow here is fine, for now, but we should tag it with a FIXME pointing at a rust-lang/rust issue, because the allow shouldn't be necessary.

Reply:
The issue has been created, see #74838.